internal/provider/remote_client.go

```diff
@@ -59,6 +61,26 @@ func (c *RemoteClient) WriteFileSCP(ctx context.Context, content string, path st
 	return scpClient.CopyFile(ctx, strings.NewReader(content), path, permissions)
 }
 
+func (c *RemoteClient) WriteFileSFTP(_ context.Context, content string, path string, permissions string) error {
```

Comment:
To make it clear that it does not set permissions?

```suggestion
func (c *RemoteClient) WriteFileSFTP(_ context.Context, content string, path string) error {
```

Reply:
There is no feature in sftp to set the permission at create file. But it's not a issue, because resourceRemoteFileCreate calls the [chmodfile](https://github.com/tenstad/terraform-provider-remote/blob/main/internal/provider/resource_remote_file.go#L136) after WriteFile. So the file permission is set.
One issue is with this: if you write sensitive content, so there maybe a small time, when the info is readable.

Reply:
I resolved it in next commit: set the permissions before write any content.